code/modules/tutorials/_tutorial.dm

```diff
@@ -0,0 +1,258 @@
+/// The base for a contextual tutorial.
+/// In order to give a tutorial to someone, use `SStutorials.suggest_tutorial(user, /datum/tutorial/subtype)`
+/datum/tutorial
+	/// If set, any account who started playing before this date will not be given this tutorial.
+	/// Date is in YYYY-MM-DD format.
+	var/grandfather_date
+
+	/// The mob we are giving the tutorial to
+	VAR_PROTECTED/mob/user
+
+	VAR_PRIVATE/atom/movable/screen/tutorial_instruction/instruction_screen
+
+/datum/tutorial/New(mob/user)
+	src.user = user
+
+	RegisterSignal(user, COMSIG_PARENT_QDELETING, PROC_REF(destroy_self))
+	RegisterSignal(user.client, COMSIG_PARENT_QDELETING, PROC_REF(destroy_self))
+
+/datum/tutorial/Destroy(force, ...)
+	user.client?.screen -= instruction_screen
+	QDEL_NULL(instruction_screen)
+
+	user = null
+
+	return ..()
+
+/// Gets the [`/datum/tutorial_manager`] that owns this tutorial.
+/datum/tutorial/proc/manager()
+	RETURN_TYPE(/datum/tutorial_manager)
+	return SStutorials.tutorial_managers[type]
+
+/// The actual steps of the tutorial. Is given any excess arguments of suggest_tutorial.
+/// Must be overridden.
+/datum/tutorial/proc/perform()
+	SHOULD_CALL_PARENT(FALSE)
+	CRASH("[type] does not override perform()")
+
+/// Returns TRUE/FALSE if this tutorial should be given.
+/// If FALSE, does not mean it won't come back later.
+/datum/tutorial/proc/should_perform()
+	SHOULD_CALL_PARENT(FALSE)
+	return TRUE
+
+/// Called by the tutorial when the user has successfully completed it.
+/// Will mark it as completed in the datbaase and kick off destruction of the tutorial.
+/datum/tutorial/proc/complete()
+	SIGNAL_HANDLER
+	PROTECTED_PROC(TRUE)
+	SHOULD_NOT_OVERRIDE(TRUE)
+
+	manager().complete(user)
+	perform_base_completion_effects()
+
+/// As opposed to `complete()`, this merely hides the tutorial.
+/// This should be used when the user doesn't need the tutorial anymore, but didn't
+/// actually properly finish it.
+/datum/tutorial/proc/dismiss()
+	SIGNAL_HANDLER
+	PROTECTED_PROC(TRUE)
+	SHOULD_NOT_OVERRIDE(TRUE)
+
+	manager().dismiss(user)
+	perform_base_completion_effects()
+
+#define INSTRUCTION_SCREEN_DELAY (1 SECONDS)
+
+/datum/tutorial/proc/perform_base_completion_effects()
+	SHOULD_NOT_OVERRIDE(TRUE)
+
+	var/delay = perform_completion_effects()
+
+	if (!isnull(instruction_screen))
+		animate(instruction_screen, time = INSTRUCTION_SCREEN_DELAY, alpha = 0, easing = SINE_EASING)
+		delay += INSTRUCTION_SCREEN_DELAY
+
+	QDEL_IN(src, delay)
+
+/// Called when the tutorial is being hidden, but before it is deleted.
+/// You should unregister signals and fade out any of your creations in here.
+/// Returns how long extra to delay the deletion.
+/datum/tutorial/proc/perform_completion_effects()
```

Comment:
would be nice if name implied the delay, what about just like perform_completion_effects_and_delay